test/test-lint.cpp

```diff
@@ -1148,6 +1148,20 @@ TEST(test_lint, assign_to_undeclared_variable) {
                             span_matcher(assignment))));
 }
 
+TEST(test_lint, use_and_assign_to_undeclared_variable) {
+  const char8 assignment[] = u8"x += 1";
```

Comment:
Nit: This string should contain just the variable's name, not an entire expression.
```suggestion
  const char8 assignment[] = u8"x";
```